scripts/acceptance-test/src/main.rs

```diff
@@ -0,0 +1,201 @@
+    slot_fetcher.subscribe_slots(false).await?;
+
+    let mut checked = 0;
+    let client = get_rollup_client()?;
+    let mut latest_batch_num = 0;
+    'outer: loop {
+        let slot = slot_fetcher.next_slot().await?.unwrap();
+        for slot_number in checked..=slot.number {
+            let Ok(snapshot) = load_snapshot_json(slot_number, &directories.snapshots_dir) else {
+                // We might be missing a few slots at the beginning.
+                // If the slot number is less than 10, just ignore the missing snapshot.
+                if slot_number < 10 {
+                    continue;
+                } else if latest_batch_num < NUM_SOAK_BATCHES {
+                    panic!("Missing snapshot for slot {}", slot_number);
+                } else {
+                    // Once we've passed NUM_SOAK_BATCHES, and we find the first missing snapshot, we're done
+                    tracing::info!(
+                        "Missing snapshot found at slot {}. Finished resyncing.",
+                        slot_number
+                    );
+                    break 'outer;
+                }
+            };
+            let slot_snapshot: Slot = serde_json::from_value(snapshot.clone()).unwrap();
+            latest_batch_num = slot_snapshot.batch_range.end.saturating_sub(1);
+            let include_children = if slot_snapshot.batches.is_empty() {
+                None
+            } else {
+                Some(GetSlotByIdChildren::_1)
+            };
+            let slot = client
+                .get_slot_by_id(&types::IntOrHash::Integer(slot_number), include_children)
+                .await?;
+            compare_against_snapshot(
+                &slot.into_inner(),
+                snapshot,
+                &format!("slot_{}", slot_number),
+                false,
+            )?;
+        }
+        checked = slot.number;
+    }
+
+    tracing::info!(
+        "Rollup resync complete. All slots match their snapshots. Found {} batches.",
+        latest_batch_num
+    );
+
+    let new_throughput_report =
+        run_soak(directories.clone(), rollup, latest_batch_num, false).await?;
+    let previous_throughput_report: ThroughputReport = serde_json::from_str::<ThroughputReport>(
+        &std::fs::read_to_string(directories.output_dir.join("throughput_report.json"))?,
+    )?;
+    let previous_throughput =
+        previous_throughput_report.num_txs as f64 / previous_throughput_report.num_slots as f64;
+    let new_throughput =
+        new_throughput_report.num_txs as f64 / new_throughput_report.num_slots as f64;
+    if new_throughput < (previous_throughput * 0.9) {
+        anyhow::bail!("Throughput is less than 90% of the previous throughput. This is likely due to a bug in the rollup. Old throughput: {:.2} txs/slot, new throughput: {:.2} txs/slot", previous_throughput, new_throughput);
+    }
+
+    // Save throughput report to acceptance test directory
```

Comment:
Should we always save the new throughput, or take max(new, old)? If we always overwrite it, it's possible for performance to slowly degrade by less than 10% each run but steadily becoming worse, i.e. several small regressions would be accepted. If we take the max then we commit to our peak throughput as our target